### PR TITLE
Multiple fixes around commitments and group keys

### PR DESCRIPTION
**Pull request by guggero**
This is a collection of bug fixes and the corresponding integration tests that discovered them.

Depends on https://github.com/lightninglabs/taproot-assets/pull/592.

Fixes https://github.com/lightninglabs/taproot-assets/issues/582.
Fixes https://github.com/lightninglabs/taproot-assets/issues/571.
Fixes https://github.com/lightninglabs/taproot-assets/issues/593.

### CODE REVIEW

File: universe/syncer.go [hunk 1]
```diff
@@ -285,7 +285,7 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot BaseRoot,
 				transferLeaves[j].Leaf.Proof.BlockHeight
 		})
 
-		go fn.SendAll(fetchedLeaves, transferLeaves...)
+		fn.SendAll(fetchedLeaves, transferLeaves...)
```
Comment (Roasbeef) [inline universe/syncer.go:288]: Nice fix, makes sense.

File: rpcserver.go [hunk 2]
```diff
@@ -1997,10 +1997,17 @@ func (r *rpcServer) BurnAsset(ctx context.Context,
 			"accidental asset burns")
 	}
 
+	var groupKey *btcec.PublicKey
+	assetGroup, err := r.cfg.TapAddrBook.QueryAssetGroup(ctx, assetID)
```
Comment (Roasbeef) [inline rpcserver.go:2018]: 👍

File: tapscript/send.go [hunk 3]
```diff
@@ -825,10 +827,6 @@ func CreateOutputCommitments(inputTapCommitments tappsbt.InputCommitments,
 				return nil, err
 			}
 
-			log.Tracef("Adding %d passive assets to output with %d "+
```
Comment (Roasbeef) [inline tapscript/send.go:828]: Seems useful to still keep on this log line about passive assets?

File: tapscript/send.go [hunk 4]
```diff
@@ -502,17 +502,18 @@ func PrepareOutputAssets(ctx context.Context, vPkt *tappsbt.VPacket) error {
 		// TODO(ffranr): Add support for interactive full value multiple
 		// input spend.
 		input := inputs[0]
+		vOut := outputs[recipientIndex]
 
 		// We'll now create a new copy of the old asset, swapping out
 		// the script key. We blank out the tweaked key information as
 		// this is now an external asset.
-		outputs[recipientIndex].Asset = input.Asset().Copy()
-		outputs[recipientIndex].Asset.ScriptKey = outputs[0].ScriptKey
```
Comment (Roasbeef) [inline tapscript/send.go:508]: As for burns we always have the burn output be second?

Comment (Roasbeef) [reply tapscript/send.go:508]: For sake of time, can you add some more detail to the commit body re the scenario?

File: tapscript/send.go [hunk 5]
```diff
@@ -791,7 +791,7 @@ func CreateOutputCommitments(inputTapCommitments tappsbt.InputCommitments,
 
 		// The output that houses the split root will carry along the
 		// existing Taproot Asset commitment of the sender.
-		if vOut.Type.IsSplitRoot() {
+		if vOut.Type.IsSplitRoot() || vOut.Type.CanCarryPassive() {
```
Comment (Roasbeef) [inline tapscript/send.go:800]: Also update the comment here?

File: tapscript/send.go (hunk 4 above)
Comment (guggero) [reply tapscript/send.go:508]: Makes sense, added more context to the commit message.

File: tapscript/send.go [hunk 6]
```diff
@@ -573,6 +576,9 @@ func PrepareOutputAssets(ctx context.Context, vPkt *tappsbt.VPacket) error {
 
 		vOut.Asset = &splitAsset.Asset
 		vOut.Asset.ScriptKey = vOut.ScriptKey
+
+		// Adjust the version for the requested send type.
+		vOut.Asset.Version = vOut.AssetVersion
```
[the remark this answers is not in the document]
Comment (guggero) [reply tapscript/send.go]: True. So I guess it's only needed in the full value send case. Removed the second one.

**Review by jharveyb — APPROVED** [review]
Looks good, great fixes 💯

**Pull request merged**